au/code/au/quantity.hh

```diff
@@ -115,13 +115,8 @@ constexpr T as_raw_number(T x) {
 }
 
 namespace detail {
-enum class UnitSign {
-    POSITIVE,
-    NEGATIVE,
-};
-
-template <typename Rep, UnitSign>
-struct CompareUnderlyingValues;
+template <typename SignMag, typename Op>
+struct SignAwareComparison;
```

Comment:
Seems like we could use some internal documentation of why we need this?  Something that points out that our comparisons happen in a common type, which looses the signed/unsigned of underlying types, and this effectively allows to implement something equivalent to `std::cmp_less` and friends?

Reply:
Added a doc comment.